### DISCUSSION TITLE
single-SPA / micro frontend / no index.html

### DISCUSSION THREAD

**Discussion by PaulKujawa** [jaredpalmer/razzle#1480] [Q&A] [answered]
Hey folks, I am trying to setup [Single-SPA](https://github.com/single-spa/single-spa) with Razzle. I got Single-SPA working with their npx, that creates a plain [React application](https://github.com/PaulKujawa/sspa-navbar) with a prepared webpack config. Now, I want to replace such an application with Razzle.

To get it working with Razzle, one needs:
* to remove the project's index.html as there is one shared by all projects
* adjust Webpack configuration
* not render the app anywhere but expose some lifecycle functions for Single-SPA to call

So I did:
* [remove SSR](https://github.com/PaulKujawa/sspa-dashboard/commit/a711b722e006b9405bcf55f3bff413aa5080e3cb) for now
* [removed the index.html](https://github.com/PaulKujawa/sspa-dashboard/commit/a024e4f45652b55ae48c6989cebcc8f79cead40d) and configured webpack according to [their Docs](https://single-spa.js.org/docs/recommended-setup#build-tools-webpack--rollup)

What I face:
* even in SPA-mode, Razzle's Webpack throws when it does not find an index.html
* when I keep the index.html, both the demo SVG file as well as the webSocket connection returns this html content

I know it's a broad topic, but perhaps someone is more familiar with SystemJS, Razzle, or Webpack :)
And no, Single-SPA has no experience with Razzle.

**Comment by fivethreeo** [comment, marked as answer]
npx create-razzle-app --example basic-spa myapp
